### PR TITLE
refactor(semver): make `difference()` more readable

### PR DESCRIPTION
**Pull request by timreichen**

### CODE REVIEW

File: semver/difference.ts [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
-/** Returns difference between two versions by the release type, or
- * `undefined` if the versions are the same. */
+/**
+ * Returns difference between two versions by the release type, or `undefined` if the versions are the same.
+ */
 export function difference(
-  s0: SemVer,
-  s1: SemVer,
+  version1: SemVer,
+  version2: SemVer,
```
Comment (iuioiua) [inline semver/difference.ts:10]: IMO, `s0` and `s1` were fine as they were. Was there something about them that was unclear?

Comment (timreichen) [reply semver/difference.ts:10]: I think we should not have to abbreviate parameter names. While the code gets a bit bigger, full names are easier to comprehend in context imo.

Comment (kt3k) [reply semver/difference.ts:10]: `s0` `s1` look fine to me too as the scope of them are enough short here. I think there's very little chance for them to be confused with something else.

Comment (timreichen) [reply semver/difference.ts]: ok, reverted.

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**